fetcher/fetcher.go

```diff
@@ -61,24 +66,34 @@ func NewPool(limit, timeout int, repos []string) Pool {
 	return f
 }
 
-func (p *Pool) Fetch(url string) (io.ReadCloser, error) {
-	client := &http.Client{
-		Timeout: time.Duration(p.timeout) * time.Second,
-	}
-	resp, err := client.Get(url)
-	if err != nil {
-		return nil, err
-	}
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to fetch with: %d", resp.StatusCode)
+func (p *Pool) RetryFetch(url string) (io.ReadCloser, error) {
+	var resp http.Response
+	for r := 1; r <= p.retries; r++ {
+		client := &http.Client{
+			Timeout: time.Duration(p.timeout) * time.Second,
+		}
+		resp, err := client.Get(url)
+		if err != nil {
+			return nil, err
+		}
+		if resp.StatusCode == http.StatusOK {
+			return resp.Body, nil
+		}
+		/* 404 must mean it doesn't exist. */
+		if resp.StatusCode == http.StatusNotFound {
+			break
+		}
```

Comment:
Do you want to tell other routines that this URL gives you a 404? Should this URL be ignored if found again?

I think this functionality already exists in this code, does the application record the URL before or after the URL has been parsed?

Reply:
* the error is returned below
* the caching of urls is done in the finder
* record how and what for?

Reply:
Just to prevent another routine from hitting the 404.

Reply:
Right, that's already handled by:
https://github.com/status-im/go-maven-resolver/blob/64a31400ba0a4950392b8fb0aca2ab6b8220cfd0/finder/finder.go#L99-L102
So we're good.